### PR TITLE
System.Data.SqlClient: Remove unnecessary LocaleMapper

### LINKED ISSUE #16709: Cleanup System.Data's LocaleMapper
`System.Data` has an internal `LocaleMapper` helper class that maps LCIDs to locale names, locale names to LCIDs, and locale names to ANSI code pages, which was necessary to workaround missing `CultureInfo` APIs in .NET Core 1.0/1.1:

 - [LocaleMapper.Unix.cs](https://github.com/dotnet/corefx/blob/b70e5d1d7485b26dfc305597148bb2b4e11f3741/src/Common/src/System/Data/Locale/LocaleMapper.Unix.cs)
 - [LocaleMapper.Windows.cs](https://github.com/dotnet/corefx/blob/b70e5d1d7485b26dfc305597148bb2b4e11f3741/src/Common/src/System/Data/Locale/LocaleMapper.Windows.cs)

However, the missing `CultureInfo` APIs were added back last summer/fall as part of the .NET Standard 2.0 effort (including [Unix support](https://github.com/dotnet/coreclr/blob/a2c684d4094d00c715d67238cf4f4650248d4ded/src/mscorlib/src/System/Globalization/LocaleData.Unix.cs)), which should allow `System.Data`'s `LocaleMapper` to be deleted or significantly pared-down.

cc: @saurabh500, @tarekgh

### PR DESCRIPTION
**Pull request by justinvp**
The internal `LocaleMapper` helper class was necessary for .NET Core 1.x due to missing `CultureInfo` APIs. Now that the missing `CultureInfo` APIs have been added back to .NET Core, it can be deleted.

Fixes #16709

### CODE REVIEW

File: src/Common/src/System/Data/Locale/Locale.cs [hunk 1]
```diff
@@ -1,58 +0,0 @@
...
-namespace System.Data
-{
-    internal static class Locale
-    {
-        // Maps between LCIDs and LocaleName+AnsiCodePages+Encodings
-        private static readonly ConcurrentDictionary<int, Tuple<string, int, Encoding>> s_cachedEncodings = new ConcurrentDictionary<int, Tuple<string, int, Encoding>>();
-
-        static Locale()
-        {
-            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
```
Comment (justinvp) [inline src/Common/src/System/Data/Locale/Locale.cs]: I was wondering if we'd need to find another place to register the `CodePagesEncodingProvider` provider, but it's already being registered in [`TdsParser`](https://github.com/dotnet/corefx/blob/562942307a5d2c771d6ff9c14a87a13e40c59b26/src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs#L34), the only class in `System.Data.SqlClient` where `Encoding.GetEncoding(int)` is called.

File: src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs [hunk 2]
```diff
@@ -3433,7 +3433,7 @@ internal int GetCodePage(SqlCollation collation, TdsParserStateObject stateObj)
                         case 0x827:     // Mapping Non-supported Lithuanian code page to supported Lithuanian.
                             try
                             {
-                                codePage = Locale.GetCodePageForLcid(0x427);
+                                codePage = new CultureInfo(0x427).TextInfo.ANSICodePage;
```
Comment (justinvp) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs]: I based the changes in this file on the [reference source](https://github.com/Microsoft/referencesource/blob/cf5e702dcbf0fd1f5f5df75b3166345591a5e98a/System.Data/System/Data/SqlClient/TdsParser.cs#L3655-L3718).

**Comment by justinvp** [comment]
@dotnet-bot test Innerloop Windows_NT Release Build and Test please (Timed out after 60000ms waiting for remote process)

**Comment by justinvp** [comment]
cc: @saurabh500 (@tarekgh, FYI)

**Comment by saurabh500** [comment]
Thanks for taking care is this @Justinvp LGTM.

File: src/Common/src/System/Data/Locale/Locale.cs (hunk 1 above)
Comment (saurabh500) [inline src/Common/src/System/Data/Locale/Locale.cs]: This warrants more discussion. @tarekgh also had some concerns about this. Can you open an issue for this?

**Pull request merged**